jdbc/src/java/org/apache/hive/jdbc/HiveStatement.java

```diff
@@ -197,6 +198,12 @@ private void closeStatementIfNeeded() throws SQLException {
     } catch (SQLException e) {
       throw e;
     } catch (Exception e) {
+      if (e instanceof TApplicationException && ((TApplicationException)e)
+          .getType() == TApplicationException.BAD_SEQUENCE_ID) {
+        String errorMsg = "Mismatch thrift sequence id, which usually is caused by an OutOfMemoryError " +
+            "or a SocketTimeoutException, please check the memory status and previous calls.";
+        throw new SQLException(errorMsg, "08S01", e);
```

Comment:
Please do not say 'is usually caused by'.  We do not yet know what "usual" is.

Try something like:

`
Mismatch thrift sequence id.  A previous call to the Thrift library failed and now position within the input stream is lost.  Please enable verbose error logging and check the status of previous calls.
`

Reply:
Thanks @belugabehr, updated the error messages.